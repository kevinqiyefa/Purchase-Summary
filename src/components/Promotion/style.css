.Promotion-Section-Title {
  display: flex;
  align-items: center;
  cursor: pointer;
}

.Promotion-Main h4 {
  text-decoration: underline;
  margin: 0.8em 0;
}

.promotion-form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
}

.promotion-form > div {
  display: flex;
  justify-content: space-between;
}

#promotion-form-input-label {
  color: rgba(0, 0, 0, 0.6);
  font-size: 0.9em;
  padding-left: 0.5em;
  font-weight: 500;
}

#promotion-form-input-field {
  margin-right: 2em;
  border: 1px solid gray;
}

#promotion-form-button {
  border: 1px solid rgb(104, 104, 104);
  color: black;
  padding: 2px 8px 5px 8px;
  font-weight: 700;
  cursor: pointer;
  border-radius: 25px;
}

#promotion-form-button:hover {
  background: #2196f3;
  border: 1px solid #2196f3;
  color: white;
}
.promotion-form span {
  color: red;
  font-size: 0.7em;
}
