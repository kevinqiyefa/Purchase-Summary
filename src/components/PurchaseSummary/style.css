.Purchase-Summary-Main {
  border: solid #dcdcdc 1px;
  border-bottom-left-radius: 8px;
  border-bottom-right-radius: 8px;
  border-top: none;
  box-sizing: border-box;
  padding: 1em;
  margin: 0 10%;
  text-align: left;
  font-size: 0.8em;
  font-family: BogleWeb, Helvetica Neue, Helvetica, Arial, sans-serif;
  box-shadow: 0 4px 2px -2px lightgrey;
}

.Purchase-Summary-Loader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid #419bee;
  width: 80px;
  height: 80px;
  -webkit-animation: spin 0.5s linear infinite; /* Safari */
  animation: spin 0.5s linear infinite;
  margin: 5em auto;
}

.bold-text {
  font-weight: bolder;
}

.red-text {
  color: red;
}

.big-text {
  font-size: 1.5em;
}

hr {
  width: 100%;
  background-color: #eee;
  border: 0 none;
  height: 1px;
  margin-top: 0.5em;
}

/* Safari */
@-webkit-keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
