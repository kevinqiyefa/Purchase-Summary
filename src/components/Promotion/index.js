import React, { Component } from 'react';
import './style.css';

class Promotion extends Component {
  state = {
    showForm: false,
    promoCode: ''
  };

  handleClick = () => {
    this.setState({ showForm: !this.state.showForm });
  };

  handlePromoCodeChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleApplyPromoCode = () => {
    //you can only apply the promo code once
    //should add a temp state to check when should display error messages
    if (
      this.state.promoCode === this.props.promoCode &&
      !this.props.discountApplied
    ) {
      this.setState({ promoCode: '', showForm: false });

      this.props.applyPromoCode();
    } else {
      this.setState({ promoCode: '' });
    }
  };

  render() {
    const errorMessage = this.props.discountApplied ? (
      <span>You can only apply once</span>
    ) : null;

    const promotionForm = (
      <form className="promotion-form">
        <label htmlFor="input-promo-code" id="promotion-form-input-label">
          Promo code
        </label>
        <div>
          <input
            type="text"
            name="promoCode"
            id="promotion-form-input-field"
            onChange={this.handlePromoCodeChange}
            value={this.state.promoCode}
          />
          <input
            type="button"
            value="Apply"
            id="promotion-form-button"
            onClick={this.handleApplyPromoCode}
          />
        </div>
        {errorMessage}
      </form>
    );

    let promoSection = this.state.showForm ? (
      <div>
        <div onClick={this.handleClick} className="Promotion-Section-Title">
          <h4>Hide promo code </h4> <span className="minus" />
        </div>

        {promotionForm}
      </div>
    ) : (
      <div onClick={this.handleClick} className="Promotion-Section-Title">
        <h4>Apply promo code </h4> <span className="add" />
      </div>
    );
    return <div className="Promotion-Main">{promoSection}</div>;
  }
}

export default Promotion;
