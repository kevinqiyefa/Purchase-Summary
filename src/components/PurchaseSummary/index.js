import React, { Component } from 'react';

import PriceSummary from '../PriceSummary';
import Promotion from '../Promotion';
import ItemDetails from '../ItemDetails';
import './style.css';

class PurchaseSummary extends Component {
  applyPromoCode = () => {
    const { subtotal, savings, tax } = this.props.pricingData.pricing;
    this.props.applyDiscount(subtotal, savings, tax);
  };

  render() {
    const {
      pricing,
      itemDetails,
      promo_code,
      discount_applied
    } = this.props.pricingData;

    return pricing ? (
      <div className="Purchase-Summary-Main">
        <PriceSummary pricing={pricing} />
        <ItemDetails itemDetails={itemDetails} pricing={pricing} />
        <Promotion
          promoCode={promo_code}
          discountApplied={discount_applied}
          applyPromoCode={this.applyPromoCode}
        />
      </div>
    ) : (
      <div className="Purchase-Summary-Loader" />
    );
  }
}

export default PurchaseSummary;
