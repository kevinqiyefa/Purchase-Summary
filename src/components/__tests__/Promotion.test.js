import React from 'react';
import { mount } from 'enzyme';
import PurchaseSummary from '../PurchaseSummary';
import Promotion from '../Promotion';
import toJson from 'enzyme-to-json';

describe('Testing ItemDetails Component', () => {
  let wrapped;

  const pricingData = {
    pricing: {
      subtotal: 102.96,
      savings: 3.85,
      tax: 8.92,
      total: 108.03,
      zip: 85050
    },
    itemDetails: {
      item_name:
        'Essntials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red',
      quantity: 1
    },
    promo_code: 'DISCOUNT',
    discount_applied: false
  };
  beforeEach(() => {
    wrapped = mount(<PurchaseSummary pricingData={pricingData} />);
  });

  it('finds promotion section class name', () => {
    expect(
      wrapped
        .find(Promotion)
        .childAt(0)
        .hasClass('Promotion-Main')
    ).toEqual(true);
  });

  it('shows state', () => {
    expect(wrapped.find(Promotion).state('showForm')).toEqual(false);
  });

  it('shows promotion form', () => {
    wrapped.find('.Promotion-Section-Title').simulate('click');
    expect(wrapped.find(Promotion).state('showForm')).toEqual(true);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapped);
    expect(serialized).toMatchSnapshot();
  });
});
