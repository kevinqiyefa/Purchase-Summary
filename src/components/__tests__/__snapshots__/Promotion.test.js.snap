// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ItemDetails Component matches snapshot 1`] = `
<PurchaseSummary
  pricingData={
    Object {
      "discount_applied": false,
      "itemDetails": Object {
        "item_name": "Essntials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
        "quantity": 1,
      },
      "pricing": Object {
        "savings": 3.85,
        "subtotal": 102.96,
        "tax": 8.92,
        "total": 108.03,
        "zip": 85050,
      },
      "promo_code": "DISCOUNT",
    }
  }
>
  <div
    className="Purchase-Summary-Main"
  >
    <PriceSummary
      pricing={
        Object {
          "savings": 3.85,
          "subtotal": 102.96,
          "tax": 8.92,
          "total": 108.03,
          "zip": 85050,
        }
      }
    >
      <div
        className="Price-Summary-Main"
      >
        <div>
          <span>
            Subtotal
          </span>
          <span
            className="bold-text"
          >
            $
            102.96
          </span>
        </div>
        <div>
          <div
            className="Price-Summary-Savings"
          >
            Pickup savings
            <span
              className="tooltiptext"
            >
              Pickup your order in the store helps cut cost, and we pass the savings on to you.
            </span>
          </div>
          <span
            className="red-text bold-text"
          >
            -$
            3.85
          </span>
        </div>
        <div>
          <div
            className="Price-Summary-Tax-Zip"
          >
            <span>
              Est. taxes & fees
            </span>
            <span>
              (Based on 
              85050
              )
            </span>
          </div>
          <span
            className="bold-text"
          >
            $
            8.92
          </span>
        </div>
        <hr />
        <div
          className="big-text"
        >
          <span
            className="bold-text"
          >
            Est. total
          </span>
          <span
            className="bold-text"
          >
            $
            108.03
          </span>
        </div>
      </div>
    </PriceSummary>
    <ItemDetails
      itemDetails={
        Object {
          "item_name": "Essntials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
          "quantity": 1,
        }
      }
      pricing={
        Object {
          "savings": 3.85,
          "subtotal": 102.96,
          "tax": 8.92,
          "total": 108.03,
          "zip": 85050,
        }
      }
    >
      <div
        className="Item-Details-Main"
      >
        <div
          className="Item-Details-Title"
          onClick={[Function]}
        >
          <h4>
            See item details 
          </h4>
           
          <span
            className="add"
          />
        </div>
      </div>
    </ItemDetails>
    <Promotion
      applyPromoCode={[Function]}
      discountApplied={false}
      promoCode="DISCOUNT"
    >
      <div
        className="Promotion-Main"
      >
        <div
          className="Promotion-Section-Title"
          onClick={[Function]}
        >
          <h4>
            Apply promo code 
          </h4>
           
          <span
            className="add"
          />
        </div>
      </div>
    </Promotion>
  </div>
</PurchaseSummary>
`;
